suite-native/discovery/src/discoveryThunks.ts

```diff
@@ -318,19 +314,19 @@ export const startDescriptorPreloadedDiscoveryThunk = createThunk(
         if (!device) {
             return;
         }
-
+        const deviceSupportedNetworks = selectSupportedNetworks(getState());
         const enabledNetworks = areTestnetsEnabled
-            ? supportedNetworkSymbols
-            : supportedMainnetSymbols;
-        const deviceSupportedNetworks = sortNetworks(getAvailableNetworks(enabledNetworks, device));
+            ? deviceSupportedNetworks
+            : deviceSupportedNetworks.filter(network => !isTestnet(network));
+        const supportedNetworks = sortNetworks(getAvailableNetworks(enabledNetworks, device));
 
         // Start tracking duration for analytics purposes
         dispatch(setDiscoveryStartTimestamp(performance.now()));
 
         await dispatch(
             createDescriptorPreloadedDiscoveryThunk({
                 deviceState,
-                deviceSupportedNetworks,
+                deviceSupportedNetworks: supportedNetworks,
```

Comment:
nit: Could you please improve the naming little bit? It is confusing to me. The value is originally stored in `const deviceSupportedNetworks`, then it is filtered out and the value is stored to `const enabledNetworks`.  `enabledNetworks` is filtered again and result stored in `supportedNetworks`. And in the end it is again mapped to the `deviceSupportedNetworks` prop at this line. 

What about using `pipe()` form ts-belt, in a way (just example):

```ts
   const deviceSupportedNetworks = pipe(selectDeviceSupported, filterTestnets, filterOutUnavailableAndBlacklisted)
```

Reply:
What about this way?

93d58d9770

Reply:
Yes, great 👍. Thanks for reworking this.